src/win/fs.c

```diff
@@ -97,13 +97,14 @@
     return;                                                                 \
   }
 
-#define MILLION ((int64_t) 1000 * 1000)
-#define BILLION ((int64_t) 1000 * 1000 * 1000)
+#define NSEC_PER_TICK 100
+#define TICKS_PER_SEC ((long)1e9 / NSEC_PER_TICK)
+static const int64_t WIN_TO_UNIX_TICK_OFFSET = 11644473600 * TICKS_PER_SEC;
```

Comment:
```suggestion
static const int64_t WIN_TO_UNIX_TICK_OFFSET = 11644473600LL * TICKS_PER_SEC;
```
Otherwise it overflows; the promotion to int64_t is done _after_ multiplication.

You cast to `long` on line 101 but `sizeof(long) < sizeof(int64_t)` on Windows, IIRC. If you cast to `int64_t` instead, multiplication would work okay because then 11644473600 gets implicitly promoted to `int64_t` too.

Reply:
The literal `11644473600` is already bigger than LONG_MAX (Which on windows is 2,147,483,647).
It needs about 35 bits for a signed representation. This means that on those platforms where long is 32 bit's, the literal should have long long type, as that is the next rank, and the smallest that can represent it.

For reference of how this works, see [6.4.4.1]/5 from the C99 draft standard: https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf
```
The type of an integer constant is the first of the corresponding list in which its value can
be represented
```

And the corresponding list in the table for an unsiffixed decimal constant:
```
int
long int
long long int
```